addons/web/static/src/core/domain.js

```diff
@@ -422,8 +422,8 @@ function matchDomain(record, domain) {
     const reversedDomain = Array.from(domain).reverse();
     const condStack = [];
     for (const item of reversedDomain) {
-        if (item in operators) {
-            const operator = operators[item];
+        const operator = typeof item === "string" && operators[item];
+        if (operator) {
```

Comment:
Conceptually strange that `in` shows worst performance than `typeof` and `[]`. Probably has to do with `in` not being used that often in most JS codebase so not optimised as it could in JS engines like v8.

Reply:
The most important part is checking the type, the `[]` part is just to avoid reading twice, I don't think it's very significant.